### ISSUE TITLE
PAMAuthenticator shouldn't normalize usernames

### ISSUE THREAD

**Issue by christiaanjs** [jupyterhub/jupyterhub#1671]
When I try to authenticate with PAM with a mixed-case username (in this case it is `swanecAdm`) I get this error:
```
    Traceback (most recent call last):
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/tornado/web.py", line 1512, in _execute
        result = yield result
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/handlers/login.py", line 94, in post
        yield self.spawn_single_user(user)
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/handlers/base.py", line 475, in spawn_single_user
        yield gen.with_timeout(timedelta(seconds=self.slow_spawn_timeout), finish_spawn_future)
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/handlers/base.py", line 445, in finish_user_spawn
        yield spawn_future
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/user.py", line 439, in spawn
        raise e
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/user.py", line 378, in spawn
        ip_port = yield gen.with_timeout(timedelta(seconds=spawner.start_timeout), f)
      File "/usr/lib/anaconda3/lib/python3.6/types.py", line 248, in wrapped
        coro = func(*args, **kwargs)
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/spawner.py", line 968, in start
        env = self.get_env()
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/spawner.py", line 960, in get_env
        env = self.user_env(env)
      File "/usr/lib/anaconda3/lib/python3.6/site-packages/jupyterhub/spawner.py", line 947, in user_env
        home = pwd.getpwnam(self.user.name).pw_dir
    KeyError: 'getpwnam(): name not found: **swanecadm'**
```

This is caused by the case normalization in [`Authenticator.normalize_username`](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/auth.py#L189).

As a hack, commenting out this line fixes the problem and I can login. Maybe `PAMAuthenticator` should override this behaviour. I am happy to have a go at fixing this myself.

**Comment by minrk** [comment]
It does need to normalize usernames to ensure they are correct for use with the Hub. However, you are right that LocalProcessSpawner does also need to be able to recover the 'true' unix name from the jupyterhub username for later calls like this, when the jupyterhub username and the local user may not be identical. We need something like `denormalize`, or perhaps more explicitly `unix_name` in cases like this.

**Comment by svzdvdoptum** [comment]
What do you mean @minrk the Hub is case insensitive when handling the usernames?

**Comment by Zsailer** [comment]
@minrk How would you go about "denormalizing" the Hub name? Are you thinking the "true" unix name would be stored in jupyterhub's database next to the normalized Hub name?

**Comment by minrk** [comment]
Closed by #2397

**Comment by soerendip** [comment]
Why was this closed? Is this resolved?
